recipes/renode-cores/helpers/renode_build_with_cmake.ps1

```diff
@@ -0,0 +1,86 @@
+$ErrorActionPreference = "Stop"
+
+# Get logical processors count
+$cpuCount = (Get-CimInstance Win32_Processor).NumberOfLogicalProcessors
+
+# Set environment variables (using robust expansion)
+$SRC_DIR = & cmd.exe /c echo %SRC_DIR%
+$PREFIX = & cmd.exe /c echo %PREFIX%
+$PKG_NAME = & cmd.exe /c echo %PKG_NAME%
+
+# Check for empty environment variables
+if ([string]::IsNullOrEmpty($SRC_DIR)) { throw "SRC_DIR is empty" }
+if ([string]::IsNullOrEmpty($PREFIX)) { throw "PREFIX is empty" }
+if ([string]::IsNullOrEmpty($PKG_NAME)) { throw "PKG_NAME is empty" }
+
+# Update CMakeLists.txt (using combined git commands)
+New-Item -ItemType Directory -Path "$SRC_DIR/updates" -Force | Out-Null
+git -C "$SRC_DIR/updates" clone "https://github.com/renode/renode-infrastructure.git" -q
+git -C "$SRC_DIR/updates/renode-infrastructure" checkout 3fc2d5fe643068e595e875d9408cb4329522b229 -q
+Copy-Item -Path "$SRC_DIR/updates/renode-infrastructure/src/Emulator/Cores/CMakeLists.txt" -Destination "$SRC_DIR/src/Infrastructure/src/Emulator/Cores/CMakeLists.txt" -Force
+
+git -C "$SRC_DIR/updates" clone "https://github.com/antmicro/tlib.git" -q
+git -C "$SRC_DIR/updates/tlib" checkout 69fff75a0eba7471283b0b8db2e55e8388e284f6 -q
+Copy-Item -Path "$SRC_DIR/updates/tlib/CMakeLists.txt" -Destination "$SRC_DIR/src/Infrastructure/src/Emulator/Cores/tlib" -Force
+Copy-Item -Path "$SRC_DIR/updates/tlib/tcg/CMakeLists.txt" -Destination "$SRC_DIR/src/Infrastructure/src/Emulator/Cores/tlib/tcg" -Force
+Copy-Item -Path "$SRC_DIR/updates/tlib/LICENSE" -Destination "$Env:RECIPE_DIR/tlib-LICENSE" -Force
+
+Copy-Item "$SRC_DIR/src/Infrastructure/src/Emulator/Cores/tlib/softfloat-3/COPYING.txt" "$Env:RECIPE_DIR/softfloat-3-COPYING.txt" -Force
+
+# Check weak implementations (using combined path)
+pushd $SRC_DIR/tools/building
+    & bash.exe -c ". './check_weak_implementations.sh'"
+popd
+
+$env:PATH = "${env:BUILD_PREFIX}/Library/mingw-w64/bin;${env:BUILD_PREFIX}/Library/bin;${env:PREFIX}/Library/bin;${env:PREFIX}/bin;${env:PATH}"
+$env:CXXFLAGS = "$env:CXXFLAGS -Wno-unused-function -Wno-maybe-uninitialized -Wno-error"
+$env:CFLAGS = "$env:CFLAGS -Wno-unused-function -Wno-maybe-uninitialized -Wno-error"
```

Comment:
It would be better if we could explicitly disable warnings rather than setting `-Wno-error`.  Otherwise this can let problematic issues slip through.

Reply:
That is why I was removing the -fPIC, this is the issue, on windows the libraries are shared, so there is a warning saying so, which is automatically turned into a failure. I had the same concern

Reply:
That is a left-over, I actually still remove the -fPIC manually. Also, I actually ran this recipe on an osx-arm64 runner, but we can revisit when adding the archs after release to feedstock